src/effects/VST/VSTEffect.cpp

```diff
@@ -987,9 +987,7 @@ size_t VSTEffect::GetBlockSize() const
 
 sampleCount VSTEffect::GetLatency() const
 {
-   if (mUseLatency)
-      return mBufferDelay;
-   return 0;
+   return GetLatencyB();
```

Comment:
`VSTEffectInstance` will also need to define all the other pure virtuals that were added to `VSTEffectWrapper`.

Some of the new functions should be empty stubs, others should have the lines from the corresponding VSTEffect functions moved into them, and data members should be moved too as needed to compile it.

Or a third possibility is that some members should be moved into VSTWrapper and the functions become non-virtual.

Which of the new functions of `VSTEffectInstance` should be empty?  I think NeedIdle, UpdateDisplay, SizeWindow all serve UI purposes only so should be empty.

What about the rest of them?  I'm not decided yet.

Reply:
The virtual functions with non-void return are `GetTimeInfo` and `GetSampleRate`.  If we have a vst handle for the plug-in distinct from the instances, will it need to report meaningful values out of `AudioMaster` or will those cases never be invoked?

I'm not sure of the answer.  But I suggest this, and let experiment see if there is a problem:

Make default implementations of these non-pure virtuals to return nullptr at 44100.

Make overrides in the Instance class.

Move the member mTimeInfo to the instance.

Reply:
It's unclear what `GetProcessLevel` is supposed to report to the plug-in, but the legacy code in fact never returned anything but 1 because mProcessLevel is never changed.

I say move mProcessLevel into the Wrapper class and provide a default implementation to the non-pure virtual.  Maybe we will figure out the need for an override later.

Reply:
I think the `void`-returning `SetBufferDelay` and `Automate` are meaningful only for instances.  Make default implementations that do nothing.  Override them in the instance.

Reply:
Automate also references mMaster, so to have it a member of the instance, I had to move it to the wrapper.

I think I implemented all your suggestions here? please check commit "4.3 Implement some virtuals in the instance"

Reply:
You did not remove `GetLatencyB` or undo the move of the two data members `mUseLatency`, `mBufferDelay`

Reply:
If you examine how mMaster is used, it's really just a boolean that is true when an instance (a VSTEffect) is NOT the master.

Reply:
You need mMaster in Automate, but the wrapper has a no-op override of Audomate, so there should be no need to put mMaster in the Wrapper.  You also moved mTimeInfo into the Wrapper but it should be in the Instane.

Reply:
> You need mMaster in Automate, but the wrapper has a no-op override of Automate, so there should be no need to put mMaster in the Wrapper.

but mMaster is set and used in 3 different VSTEffect methods - that's why I put it in the wrapper. I probably could rework the code in a way that VSTEffect does not need it anymore, but I'd leave that for later

Reply:
> You also moved mTimeInfo into the Wrapper but it should be in the Instance.

`mTimeInfo` is again written/used in different methods of `VSTEffect`. Should I figure out how to move these uses of it from `VSTEffect` to `VSTEffectInstance` as part of step 4?